p2p/proxy.go

```diff
@@ -112,11 +112,8 @@ func (p *Proxy) Start() error {
 	go p.read(p.Peer2, p.Peer1, errorChannel)
 
 	if p.Peer2.handshakeInfo != nil {
-
-		err := triggerHandshake(p.Peer2)
-		if err != nil {
-			return fmt.Errorf("connect and start: trigger handshake: %s", err)
-		}
+		return errors.Wrap(triggerHandshake(p.Peer2),
```

Comment:
Revert to old style.

Reply:
In fact, this was wrong, before, the inner `if` was doing a return only when an error would occur. Now, the return happens whether there is an error or not, which is not the same as before.